test/app-tap/module_api.c

```diff
@@ -2335,6 +2335,80 @@ test_tuple_validate_formatted(lua_State *L)
 	return 1;
 }
 
+static int
+session_check_admin(lua_State *L)
+{
+	int sess_id = box_session_id();
+	bool sess_exist = box_session_exists(sess_id);
+	assert(sess_exist);
+
+	sess_id = box_session_id();
+	char peer_str_arg[100];
+	const char *peer_str = box_session_peer(sess_id, peer_str_arg, 100);
+    bool peer_is_null = (peer_str == NULL);
+	assert(peer_is_null);
+
+	bool sess_not_exist = !box_session_exists(1234567890);
+	assert(sess_not_exist);
+
+	lua_pushboolean(L, 1);
+
+	return 1;
+}
+
+static int
+fiber_session_check_func(va_list va)
+{
+	(void) va;
+	do {
+		fiber_set_cancellable(true);
+		fiber_sleep(0.01);
+		if (fiber_is_cancelled()) {
+			int sess_id = box_session_id();
+			if (sess_id == 0) {
```

Comment:
This is a really strange test. Why is 0 session ID considered a fail? Maybe better return the session ID via out parameters and in the test function check it does not equal to the current session?

Reply:
This part of tests is copy of https://github.com/tarantool/tarantool/blob/master/test/box-tap/session.test.lua#L31-L35
The logic of test is fully borrowed from there.